subsys/net/lib/coap/coap.c

```diff
@@ -763,6 +763,19 @@ static bool uri_path_eq(const struct coap_packet *cpkt,
 			continue;
 		}
 
+#if defined(CONFIG_COAP_URI_WILDCARD)
```

Comment:
Preferred way to check this would be like this
```
if (IS_ENABLED(CONFIG_COAP_URI_WILDCARD) && strlen(path[j]) == 1) {
....
}
```
This way the compiler will always check the compilation and it will not bitrot. There is a minimal chance for that here as the symbol is enabled by default but it is a good habit to write this check like this.